### PR TITLE
pacemaker_remote enhancements

### PR DESCRIPTION
**Pull request by kgaillot**
Most of this is refactoring to prepare for tracking remote node status. Also, some CTS and regression test issues are fixed, and the Pacemaker Remote documentation's KVM walk-through is updated.

### CODE REVIEW

**Comment by kgaillot** [comment]
FYI removed a couple of commits from this request b/c those areas might see more changes, and to trigger travis again since ubuntu was flaky the first time.

**Comment by beekhof** [comment]
holy heck

File: crmd/te_utils.c [hunk 1]
```diff
@@ -31,7 +31,96 @@
...
+
+        crm_notice("Marking %s, target of a previous stonith action, as clean", target);
+        send_stonith_update(NULL, target, uuid);
+        free(target);
+    }
+    g_list_free(stonith_cleanup_list);
+    stonith_cleanup_list = NULL;
+}
+
+/* end stonith cleanup list functions */
```
Comment (beekhof) [inline crmd/te_utils.c]: Now I understand why you went to the effort of making all these functions. Definitely helps having them all in one place :)

File: cts/CTStests.py [hunk 2]
```diff
@@ -1693,6 +1693,19 @@ def __init__(self, cm):
...
+        return is_managed == "true"
+
+    def _set_unmanaged(self, node):
+        self.debug("Disable resource management")
+        self.rsh(node, "crm_attribute -t rsc_defaults -n is-managed -v false")
+
+    def _set_managed(self, node):
+        self.debug("Re-enable resource management")
+        self.rsh(node, "crm_attribute -t rsc_defaults -n is-managed -D")
+
```
Comment (beekhof) [inline cts/CTStests.py]: ahhahaha.  we came up with the same commits.
did you ever figure out why the old code stopped working? (i tried, failed, fixed it in much the same way you did)

Comment (kgaillot) [inline cts/CTStests.py]: It was relying on the crm_attribute command line being logged, which went away as part of the password hiding stuff. Luckily this was the only spot in CTS that was looking for a command line :)

**Pull request merged**